discovery_engine_core/core/src/document.rs

```diff
@@ -63,23 +79,86 @@ pub struct Document {
     /// Stack from which the document has been taken.
     pub stack_id: StackId,
 
-    /// Position of the document from the source.
-    pub rank: usize,
+    /// Embedding from smbert.
+    pub smbert_embedding: Embedding1,
+
+    /// Resource this document refers to.
+    pub web_resource: WebResource,
+}
 
-    /// Text title of the document.
+/// Represents different kinds of resources like web, image, video, news,
+/// that are delivered by an external content API.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct WebResource {
```

Comment:
Can you elaborate how this structure would be extended to encompass e.g. video-related information such as its length, likes, etc? Is this something that would be added to the `WebResourceProvider`?

Reply:
It was supposed to be more general but we support only news. I renamed this to `NewsResource`.